[Search] Skipped FTR fixes & clean-up

TattdCodeMonkey opened the pull request:
## Summary

- Remove index details test that waits for not found empty state to refresh when index is created. Test is consistently flakey and not really that beneficial to test.
- Removed onboarding API keys tests, these rely on sessions storage which has been consistently flakey to try and rely on in FTRs
- Fixed Agents nav tests for hosted
- Moved Agents nav test to main hosted config from the feature flags
- Fixed race condition and unskipped Query Rules FTRs

Closes https://github.com/elastic/kibana/issues/237043
Closes https://github.com/elastic/kibana/issues/236363
Closes https://github.com/elastic/kibana/issues/227104
Closes https://github.com/elastic/kibana/issues/227730

### Checklist

- [ ] Any text added follows [EUI's writing guidelines](https://elastic.github.io/eui/#/guidelines/writing), uses sentence case text and includes [i18n support](https://github.com/elastic/kibana/blob/main/src/platform/packages/shared/kbn-i18n/README.md)
- [ ] [Documentation](https://www.elastic.co/guide/en/kibana/master/development-documentation.html) was added for features that require explanation or tutorials
- [x] [Unit or functional tests](https://www.elastic.co/guide/en/kibana/master/development-tests.html) were updated or added to match the most common scenarios
- [ ] If a plugin configuration key changed, check if it needs to be allowlisted in the cloud and added to the [docker list](https://github.com/elastic/kibana/blob/main/src/dev/build/tasks/os_packages/docker_generator/resources/base/bin/kibana-docker)
- [ ] This was checked for breaking HTTP API changes, and any breaking changes have been approved by the breaking-change committee. The `release_note:breaking` label should be applied in these situations.
- [ ] [Flaky Test Runner](https://ci-stats.kibana.dev/trigger_flaky_test_runner/1) was used on any tests changed
- [x] The PR  description includes the appropriate Release Notes section, and the correct `release_note:*` label is applied per the [guidelines](https://www.elastic.co/guide/en/kibana/master/contributing.html#kibana-release-notes-process)
- [x] Review the [backport guidelines](https://docs.google.com/document/d/1VyN5k91e5OVumlc0Gb9RPa3h1ewuPE705nRtioPiTvY/edit?usp=sharing) and apply applicable `backport:*` labels.

TattdCodeMonkey committed:
test(search): index details remove page load error refresh

- Closes #236363

Remove the FTR test that expects the page to refresh when a new index is
created on the index not found empty state for the index details page.
This test is flakey, and not that useful.

TattdCodeMonkey committed:
test(search): fix agents navigation tests

- Closes #237043

Updated expected link test from Agent Builder to Agents

TattdCodeMonkey committed:
test(search): move agent builder nav tests

Moved the agent builder nav tests to the main test suite instead of
feature flags since it is now on by default

TattdCodeMonkey committed:
test(search): remove onboarding api keys tests

- Closes #227104

Removing the search onboarding api key FTRs since these rely on sessions
storage which we have continuously found to be flakey and keeping these
tests passing and not skipped has proved to not be worth the effort for
the benefit they provide.

elasticmachine commented:
## :yellow_heart: Build succeeded, but was flaky
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/374260)
* Commit: 237deb8fc92a325bdf7b7acc23c7d5dcc7152e4c

### Failed CI Steps
* [Jest Tests #4](https://buildkite.com/elastic/kibana-pull-request/builds/374260#019b298a-ad88-468d-9db7-9396f6ab3ec5)

### Test Failures
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/374260#019b298a-ad88-468d-9db7-9396f6ab3ec5) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/374260/jobs/019b298a-ad88-468d-9db7-9396f6ab3ec5/artifacts/019b299e-e337-45fb-88bc-d8d35f445be6) Jest Tests #<span></span>4 / useSyntheticsRules dispatches enableDefaultAlertingSilentlyAction when canSave is true and popover opens
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/374260#019b298a-ad88-468d-9db7-9396f6ab3ec5) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/374260/jobs/019b298a-ad88-468d-9db7-9396f6ab3ec5/artifacts/019b299e-e336-49c0-aee1-32f64b8141cc) Jest Tests #<span></span>4 / useSyntheticsRules dispatches getDefaultAlertingAction when canSave is false and popover opens

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
> ✅ unchanged

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/374260","number":374260,"commit":"237deb8fc92a325bdf7b7acc23c7d5dcc7152e4c"}],"number":374260}
buildkite-pr-comment-->

TattdCodeMonkey committed:
query_rules: improve ftr

diagnosing failing FTRs for query rules, improved test to give a better
error and found the root cause, which is a race-condition on rendering
for the query rules detail panel. but leaving this skipped as I need
some assistance in resolving the underlying issue.

TattdCodeMonkey committed:
fix: query rules ftr

Updated QueryRuleDetailPanel to only default flyout open on the first
render. tracking render with state is a bit hacky and I would prefer we
refactor this out in the future, but it resolves the issue and fixes
the functional tests so we can re-enable them. We likely need better
state management for the create flow for a cleaner resolution.

Closes #227730

kibanamachine commented:
## Flaky Test Runner Stats
### 🟠 Some tests failed. - [kibana-flaky-test-suite-runner#10183](https://buildkite.com/elastic/kibana-flaky-test-suite-runner/builds/10183)
[✅] x-pack/solutions/search/test/functional_search/config.ts: 25/25 tests passed.
[❌] x-pack/solutions/search/test/serverless/functional/configs/config.search_features.ts: 0/25 tests passed.

[see run history](https://buildkite.com/elastic/kibana-flaky-test-suite-runner/builds?branch=refs%2Fpull%2F246658%2Fhead)

TattdCodeMonkey commented:
@elasticmachine merge upstream

kibanamachine commented:
## Flaky Test Runner Stats
### 🟠 Some tests failed. - [kibana-flaky-test-suite-runner#10191](https://buildkite.com/elastic/kibana-flaky-test-suite-runner/builds/10191)
[❌] x-pack/solutions/search/test/serverless/functional/configs/config.search_features.ts: 0/25 tests passed.

[see run history](https://buildkite.com/elastic/kibana-flaky-test-suite-runner/builds?branch=refs%2Fpull%2F246658%2Fhead)

wildemat committed:
Merge branch 'main' into rnorris-clean-up-skipped-ftrs

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/376737)
* Commit: 68361686d011ee2660a9b2acefcab8a73d420392

### Failed CI Steps
* [FTR Configs #27](https://buildkite.com/elastic/kibana-pull-request/builds/376737#019b51ad-90cc-4a77-8cee-5e926b175716)
* [FTR Configs #27](https://buildkite.com/elastic/kibana-pull-request/builds/376737#019b51d4-997a-432f-a68b-5e709a23c0be)
* [FTR Configs #85](https://buildkite.com/elastic/kibana-pull-request/builds/376737#019b51ad-9126-4d6d-99d7-691919989ebc)

### Test Failures
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/376737#019b51ad-90cc-4a77-8cee-5e926b175716) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/376737/jobs/019b51ad-90cc-4a77-8cee-5e926b175716/artifacts/019b51d4-6f9b-4fe9-97b2-dd1ab026d900) FTR Configs #<span></span>27 / Cloud Security Posture Verify cloud_security_posture telemetry payloads includes asset_inventory_cloud_connector_usage_stats in telemetry
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/376737#019b51d4-997a-432f-a68b-5e709a23c0be) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/376737/jobs/019b51d4-997a-432f-a68b-5e709a23c0be/artifacts/019b51dc-01f9-4920-b791-1f8337fa2330) FTR Configs #<span></span>27 / Cloud Security Posture Verify cloud_security_posture telemetry payloads includes asset_inventory_cloud_connector_usage_stats in telemetry

### History
* :broken_heart: [Build #376318](https://buildkite.com/elastic/kibana-pull-request/builds/376318) failed 68361686d011ee2660a9b2acefcab8a73d420392
* :yellow_heart: [Build #375093](https://buildkite.com/elastic/kibana-pull-request/builds/375093) was flaky 3eef77659fd8f3486cf4671629d9df951c55b066
* :yellow_heart: [Build #374260](https://buildkite.com/elastic/kibana-pull-request/builds/374260) was flaky 237deb8fc92a325bdf7b7acc23c7d5dcc7152e4c

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/376737","number":376737,"commit":"68361686d011ee2660a9b2acefcab8a73d420392"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/376318","number":376318,"commit":"68361686d011ee2660a9b2acefcab8a73d420392"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/375093","number":375093,"commit":"3eef77659fd8f3486cf4671629d9df951c55b066"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/374260","number":374260,"commit":"237deb8fc92a325bdf7b7acc23c7d5dcc7152e4c"}],"number":376737}
buildkite-pr-comment-->

wildemat committed:
Merge branch 'main' into rnorris-clean-up-skipped-ftrs

elasticmachine commented:
## :yellow_heart: Build succeeded, but was flaky
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/376776)
* Commit: 3f185c30c3d5413c2f09744406910569b3c2ed7c

### Failed CI Steps
* [Scout: [ platform / streams_app ] plugin](https://buildkite.com/elastic/kibana-pull-request/builds/376776#019b56e3-d955-4bd7-909f-eb4061003370)

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### Async chunks
> Total size of all lazy-loaded chunks that will be downloaded as the user navigates the app


| id | [before](https://github.com/elastic/kibana/commit/b1d74aab43b200d470f33f0d4d9dae7a08b4df87) | [after](https://github.com/elastic/kibana/commit/3f185c30c3d5413c2f09744406910569b3c2ed7c) | diff |
| --- | --- | --- | --- |
| `searchQueryRules` | 130.1KB | 130.2KB | +134.0B |



### History
* :broken_heart: [Build #376737](https://buildkite.com/elastic/kibana-pull-request/builds/376737) failed 68361686d011ee2660a9b2acefcab8a73d420392
* :broken_heart: [Build #376318](https://buildkite.com/elastic/kibana-pull-request/builds/376318) failed 68361686d011ee2660a9b2acefcab8a73d420392
* :yellow_heart: [Build #375093](https://buildkite.com/elastic/kibana-pull-request/builds/375093) was flaky 3eef77659fd8f3486cf4671629d9df951c55b066
* :yellow_heart: [Build #374260](https://buildkite.com/elastic/kibana-pull-request/builds/374260) was flaky 237deb8fc92a325bdf7b7acc23c7d5dcc7152e4c

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/376776","number":376776,"commit":"3f185c30c3d5413c2f09744406910569b3c2ed7c"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/376737","number":376737,"commit":"68361686d011ee2660a9b2acefcab8a73d420392"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/376318","number":376318,"commit":"68361686d011ee2660a9b2acefcab8a73d420392"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/375093","number":375093,"commit":"3eef77659fd8f3486cf4671629d9df951c55b066"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/374260","number":374260,"commit":"237deb8fc92a325bdf7b7acc23c7d5dcc7152e4c"}],"number":376776}
buildkite-pr-comment-->
